actionpack/test/controller/flash_test.rb

```diff
@@ -224,6 +224,16 @@ def test_redirect_to_with_adding_flash_types
     @controller = original_controller
   end
 
+  def test_additional_flash_types_are_not_listed_in_actions_set
+    original_controller = @controller
+    test_controller_with_flash_type_foo = Class.new(TestController) do
+      add_flash_types :foo
+    end
+    assert_not_includes test_controller_with_flash_type_foo.action_methods, "foo"
+  ensure
+    @controller = original_controller
```

Comment:
I don't think we need to backup and restore `@controller` since it is never used in this test case.